In [37]:
import pandas as pd
import numpy as np
import quandl
from sklearn.tree import DecisionTreeRegressor 
from pandas_profiling import ProfileReport

* Here we use the DecisionTreeRegressor for perform the regression funtion

In [38]:
quandl.ApiConfig.api_key = "x6fVzQf8o1joCSyhKBoY"
stock_data = quandl.get('NSE/INFIBEAM', start_date='2018-12-01',end_date='2018-12-31')
print(stock_data)

             Open   High    Low   Last  Close  Total Trade Quantity  \
Date                                                                  
2018-12-03  44.50  49.25  44.10  48.20  47.85            26696990.0   
2018-12-04  48.65  52.45  48.60  50.20  50.45            26100305.0   
2018-12-05  50.40  50.65  46.15  47.95  47.25            18384298.0   
2018-12-06  46.75  47.65  44.40  44.55  44.85            13268774.0   
2018-12-07  45.95  46.50  43.10  44.40  44.45            13889470.0   
2018-12-10  42.00  43.75  39.55  39.90  39.90            21824081.0   
2018-12-11  37.25  42.25  37.20  42.00  41.65            24656308.0   
2018-12-12  42.05  47.20  42.05  46.35  46.70            27910641.0   
2018-12-13  47.20  50.50  45.50  46.70  47.15            40448139.0   
2018-12-14  47.00  47.85  45.75  46.85  46.60            17975886.0   
2018-12-17  46.90  49.55  46.90  48.35  48.65            21539853.0   
2018-12-18  48.95  50.80  48.10  48.75  48.95            26753724.0   
2018-1

#### Now convert the data into Dataframe and then save into csv file

In [39]:
dataset = pd.DataFrame(stock_data)

In [40]:
dataset.head()

Open   High    Low   Last  Close  Total Trade Quantity  \
Date                                                                  
2018-12-03  44.50  49.25  44.10  48.20  47.85            26696990.0   
2018-12-04  48.65  52.45  48.60  50.20  50.45            26100305.0   
2018-12-05  50.40  50.65  46.15  47.95  47.25            18384298.0   
2018-12-06  46.75  47.65  44.40  44.55  44.85            13268774.0   
2018-12-07  45.95  46.50  43.10  44.40  44.45            13889470.0   

            Turnover (Lacs)  
Date                         
2018-12-03         12588.85  
2018-12-04         13315.45  
2018-12-05          8804.91  
2018-12-06          6085.51  
2018-12-07          6219.22

In [41]:
dataset.to_csv("INFIBEAM.csv")

In [42]:
data = pd.read_csv("INFIBEAM.csv")

In [43]:
data.head()

Date   Open   High    Low   Last  Close  Total Trade Quantity  \
0  2018-12-03  44.50  49.25  44.10  48.20  47.85            26696990.0   
1  2018-12-04  48.65  52.45  48.60  50.20  50.45            26100305.0   
2  2018-12-05  50.40  50.65  46.15  47.95  47.25            18384298.0   
3  2018-12-06  46.75  47.65  44.40  44.55  44.85            13268774.0   
4  2018-12-07  45.95  46.50  43.10  44.40  44.45            13889470.0   

   Turnover (Lacs)  
0         12588.85  
1         13315.45  
2          8804.91  
3          6085.51  
4          6219.22

### Now checking the null value or using pandas profiling

In [44]:
data.isnull().sum()

Date                    0
Open                    0
High                    0
Low                     0
Last                    0
Close                   0
Total Trade Quantity    0
Turnover (Lacs)         0
dtype: int64

In [45]:
ProfileReport(data)

### Note:
* now divide the data in Dependent and Independent variable

* we can see date columns in useful for our prediciton but for     simplicity we have to remove it because data format is not proper.

* Now we have to predict open price so this columns is out dependent variable because open price on High,Low, Close, Last, Turnover etc


In [46]:
data.head()

Date   Open   High    Low   Last  Close  Total_Trade_Quantity  \
0  2018-12-03  44.50  49.25  44.10  48.20  47.85            26696990.0   
1  2018-12-04  48.65  52.45  48.60  50.20  50.45            26100305.0   
2  2018-12-05  50.40  50.65  46.15  47.95  47.25            18384298.0   
3  2018-12-06  46.75  47.65  44.40  44.55  44.85            13268774.0   
4  2018-12-07  45.95  46.50  43.10  44.40  44.45            13889470.0   

   Turnover_(Lacs)  
0         12588.85  
1         13315.45  
2          8804.91  
3          6085.51  
4          6219.22

In [47]:
# let's select our features
x = data.loc[:,'High':'Turnover_(Lacs)']

In [48]:
# select the label col
y = data.loc[:,'Open']   # select all the row and "open" col only

In [49]:
x.head()

High    Low   Last  Close  Total_Trade_Quantity  Turnover_(Lacs)
0  49.25  44.10  48.20  47.85            26696990.0         12588.85
1  52.45  48.60  50.20  50.45            26100305.0         13315.45
2  50.65  46.15  47.95  47.25            18384298.0          8804.91
3  47.65  44.40  44.55  44.85            13268774.0          6085.51
4  46.50  43.10  44.40  44.45            13889470.0          6219.22

In [50]:
y.head()

0    44.50
1    48.65
2    50.40
3    46.75
4    45.95
Name: Open, dtype: float64

In [51]:
# Now split the data in training and testing

from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size = 0.1,random_state = 0)

##### Let's fit our DecisionTreeModel

In [52]:
Regressor = DecisionTreeRegressor()

##### Let;s Fit out Data

In [53]:
Regressor.fit(x_train,y_train)

DecisionTreeRegressor(ccp_alpha=0.0, criterion='mse', max_depth=None,
                      max_features=None, max_leaf_nodes=None,
                      min_impurity_decrease=0.0, min_impurity_split=None,
                      min_samples_leaf=1, min_samples_split=2,
                      min_weight_fraction_leaf=0.0, presort='deprecated',
                      random_state=None, splitter='best')

##### Let's make a prediciton on random day Data

In [54]:
test1 = [[46.50 ,43.10 ,44.40, 44.45, 13889470.0 ,6219.22]]

In [55]:
predicition = Regressor.predict(test1)

In [56]:
predicition

array([45.95])

In [57]:
test2 = [[48.05,47.00,47.35,47.45,5866161.0,2788.80]]

In [58]:
predicition = Regressor.predict(test2)

In [59]:
predicition

array([48.05])

### Conclusion:- 
on that day INFIber was open at 47.6 and our predicited value is 48.05 so we are very close at real value

In [60]:
test3 = [[46.50,45.15,45.35,45.35,5358817.0,2448.28]]

In [61]:
predicition = Regressor.predict(test3)

In [62]:
predicition

array([46.7])